src/rp2_common/hardware_dma/include/hardware/dma.h

```diff
@@ -891,6 +891,37 @@ static inline uint dma_get_timer_dreq(uint timer_num) {
     return DREQ_DMA_TIMER0 + timer_num;
 }
 
+/*! \brief Performs DMA channel cleanup after use
+ *  \ingroup hardware_dma
+ *
+ * This can be used to cleanup dma channels when they're no longer needed, such that they are in a clean state for reuse.
+ * IRQ's for the channel are disabled, any in flight-transfer is aborted and any outstanding interrupts are cleared.
+ * The channel is then clear to be reused for other purposes.
+ *
+ * \code
+ * if (dma_channel >= 0) {
+ *     dma_channel_cleanup(dma_channel);
+ *     dma_channel_unclaim(dma_channel);
+ *     dma_channel = -1;
+ * }
+ * \endcode
+ *
+ * \param channel DMA channel
+ */
+static inline void dma_channel_cleanup(uint channel) {
+    check_dma_channel_param(channel);
+    // we definitely want to disable chaining, because that could cause a restart
+    // during abort. Sicne we're doing that, it's just as easy to reset to default config!
+    uint32_t config = dma_channel_get_default_config(channel);
```

Comment:
CI is falling over because this should be `dma_channel_config config` rather than `uint32_t config` ?

Reply:
fixed